doc/api/addons.markdown

```diff
@@ -698,3 +698,57 @@ Test it with:
     var result = addon.add(obj1, obj2);
 
     console.log(result); // 30
+
+### AtExit hooks
+#### void AtExit(callback, args)
+
+* `callback`: `void (*)(void*)` - A pointer to the function to call at exit.
+* `args`: `void*` - A pointer to pass to the callback at exit.
+
+Registers exit hooks that run after the event loop has ended, but before the VM is killed.
+
+Callbacks are run in reverse order of registration, i.e. newest first. AtExit takes callback and its arguments as arguments.
+
+The file `binding.cc` implements AtExit below:
+
+      #undef NDEBUG
+      #include <assert.h>
+      #include <stdlib.h>
+      #include <node.h>
+
+      static char cookie[] = "yum yum";
+      static int at_exit_cb1_called = 0;
+      static int at_exit_cb2_called = 0;
+
+      static void at_exit_cb1(void* arg) {
+        v8::Isolate* isolate = v8::Isolate::GetCurrent();
+        v8::HandleScope scope(isolate);
+        assert(arg == 0);
+        v8::Local<v8::Object> obj = v8::Object::New(isolate);
+        assert(!obj.IsEmpty()); // assert VM is still alive
+        assert(obj->IsObject());
+        at_exit_cb1_called++;
+      }
+
+      static void at_exit_cb2(void* arg) {
+        assert(arg == static_cast<void*>(cookie));
+        at_exit_cb2_called++;
+      }
+
+      static void sanity_check(void) {
+        assert(at_exit_cb1_called == 1);
+        assert(at_exit_cb2_called == 2);
+      }
+
+      void init(Handle<Object> exports) {
+        AtExit(at_exit_cb1);
```

Comment:
Maybe `AtExit(at_exit_cb1, exports->GetIsolate()` and then `static_cast<v8::Isolate*>(arg)` in the callback?  `v8::Isolate::GetCurrent()` is going away someday so I'd prefer not to encourage its use in the documentation.

(That said, I realize it's used in plenty of other places in this document.  I'll file a PR to clean that up.)